src/wallet/wallet.cpp

```diff
@@ -1934,17 +1953,6 @@ void CWallet::ResubmitWalletTransactions(bool relay, bool force)
     // even if forcing.
     if (!fBroadcastTransactions) return;
 
-    // During reindex, importing and IBD, old wallet transactions become
-    // unconfirmed. Don't resend them as that would spam other nodes.
-    // We only allow forcing mempool submission when not relaying to avoid this spam.
-    if (!force && relay && !chain().isReadyToBroadcast()) return;
-
-    // Do this infrequently and randomly to avoid giving away
-    // that these are our transactions.
-    if (!force && GetTime() < m_next_resend) return;
-    // resend 12-36 hours from now, ~1 day on average.
-    m_next_resend = GetTime() + (12 * 60 * 60) + GetRand(24 * 60 * 60);
```

Comment:
By removing this line, calls such as `importaddress`, `importpubkey`, `importmulti` and `importdescriptors` with rescan enabled won't update the timeout because now the timeout is only reset in `MaybeResendWalletTxs` with `SetNextResend`, is this normal?

See https://github.com/bitcoin/bitcoin/blob/6610cc99a6a479ca22d272a1d583ab2a8422b7ae/src/wallet/rpc/backup.cpp#L471-L475

Reply:
Yes, this is intentional. Prior to #25768, we had two functions [`ReacceptWalletTransactions()`](https://github.com/achow101/bitcoin/blob/e191fac4f3c37820f0618f72f0a8e8b524531ab8/src/wallet/wallet.cpp#L1860) (which did not update/read `nNextResend`) and [`ResendWalletTransactions()`](https://github.com/achow101/bitcoin/blob/e191fac4f3c37820f0618f72f0a8e8b524531ab8/src/wallet/wallet.cpp#L1937) (which did update/read `nNextResend`, only called from `MaybeUpdateWalletTxs()`), which got merged into a single parameterized `ResubmitWalletTransactions()` (and updates/reads `m_next_resend`).

Since the purpose of `m_next_resend` is to prevent overly frequent _relaying_ of transactions, we should only update the timer when relaying is triggered, so this PR just undoes that behaviour change introduced in #25768.

I tried to summarize this rationale in the commit message of 3e5d3f8705154e09ecf3639f6d261595992fc5b5 - lmk if that message is unclear?

Reply:
Great, that makes sense since these RPC calls don't relay transactions. Your commit message is also clear with this in mind :+1:.